jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriBuilder.java

```diff
@@ -656,7 +656,7 @@ protected StringBuilder replaceParameter(Map<String, ? extends Object> paramMap,
 				start = matcher.end();
 				continue;
 			}
-			throw new IllegalArgumentException(Messages.MESSAGES.pathParameterNotProvided(param));
+            throw new IllegalArgumentException(Messages.MESSAGES.templateParameterNull(param));

```

Comment:
In ResteasyUriBuilderTest.testFromPathWithMissingValues the error message will be " NULL value for template parameter: w". But if we change test from 

> URI uri = ResteasyUriBuilder.fromPath("").path("{v}/{w}").build(new Object[] {"first"}, false);

to

> URI uri = ResteasyUriBuilder.fromPath("").path("{v}/{w}").build(new Object[] {"first",null}, false);

then the message will be "path param w has not been provided by the parameter map". Actually contrary to the previous case param w has been provided but with a null value so message should be " NULL value for template parameter: w" in this case and "path param w has not been provided by the parameter map" in the first case.

It's appears clearer when you change test to 

> Map<String, String> parameters=new HashMap<>(2);
	    	parameters.put("v", "first");
	    	parameters.put("w", null);
	    	URI uri = ResteasyUriBuilder.fromPath("").path("{v}/{w}").buildFromMap(parameters, false);

The error will be ""path param w has not been provided by the parameter map" it will have been better to have "NULL value for template parameter: w".

What do you think ?